app/servers.py

```diff
@@ -0,0 +1,156 @@
+"""
+Provides Server data class type for storing game
+server config info
+"""
+import traceback
+import json
+import logging
+from enum import Enum
+from marshmallow import Schema, fields, validate
+
+logging.basicConfig(level=logging.WARN)
+
+
+class ServerType(str, Enum):
+    """
+    Enum for storing supported backend server types
+    """
+    STEAM = 'STEAM'
+    DCS = 'DCS'
+    SPACE_ENGINEERS = 'SPACE_ENGINEERS'
+
+
+class Server(Schema):
+    """
+    Object that stores a given game server
+    configuration
+    """
+    name = fields.Str()
+    ip_address = fields.IP()
+    port = fields.Int(validate=validate.Range(1, 65535))
+    password = fields.Str()
+    server_type = fields.Enum(ServerType)
+
+
+server_list = {}
+
+
+def add_server(name: str, ip_address: str, port: int, server_type: str,
+               password: str = "") -> Server:
+    """
+    Adds a server to the actively monitored list
+    """
+    try:
+        if not name in server_list:
+            server_info = {
+                'name': name,
+                'ip_address': ip_address,
+                'port': port,
+                'server_type': server_type,
+                'password': password
+            }
+            server = Server().load(server_info)
+            server_list[name] = server
+            return server
+        raise ValueError("Server name already taken")
+    except Exception:
+        logging.error("Failed to add server")
+        traceback.print_exc()
+        raise
+
+
+def delete_server(name):
+    """
+    Deletes a server from the actively monitored list
+    """
+    if name == '*':
+        server_list.clear()
+    else:
+        server_list.pop(name)
+
+
+def update_server(name: str, server_info: dict):
+    """
+    Updates a server entry in the bot's monitoring list
+    Side note: this is essentially the same op as add_server
+    but without the duplicate key check to allow for overwrites
+    """
+    try:
+        schema = Server()
+        server = schema.load(server_info)
+        server_list[name] = server
+    except ValueError:
+        logging.error("Failed to update server")
+        traceback.print_exc()
```

Comment:
If we're being strict about "update", this function should check if the server exists in the first place before actually attempting to update the server. If the server doesn't exist, then the user should use the `add_server` function

Just something to think about, not necessarily a required change